file/writable_file_writer.cc

```diff
@@ -81,26 +82,74 @@ IOStatus WritableFileWriter::Append(const Slice& data) {
     assert(buf_.CurrentSize() == 0);
   }
 
-  // We never write directly to disk with direct I/O on.
-  // or we simply use it for its original purpose to accumulate many small
-  // chunks
-  if (use_direct_io() || (buf_.Capacity() >= left)) {
-    while (left > 0) {
-      size_t appended = buf_.Append(src, left);
-      left -= appended;
-      src += appended;
-
-      if (left > 0) {
-        s = Flush();
-        if (!s.ok()) {
-          break;
+  if (perform_data_verification_ && buffered_data_with_checksum_ &&
+      crc32c_checksum != 0) {
```

Comment:
Right, in the current case, if the actual checksum is 0, it will not be checked. Append is widely used nearly everywhere for write and in most cases checksum is not provided. So explicitly use a new API for non-checksum case might not be a efficient engineering change. How about we require a flag in the API: Append(const Slice& data, bool checksum_provided = false, uint32_t crc32c_checksum = 0)? When the flag is set, we trust whatever is provided in the crc32c_checksum, otherwise, checksum must be re-calcualted?